app/controllers/api/v2/application_controller.rb

```diff
@@ -5,23 +5,34 @@ class ApplicationController < ::ApplicationController
 
       def authenticate
         authenticate_or_request_with_http_token do |token|
-          if Settings.authentication.algorithm == 'plain-text'
-            # This method can be used in development mode to simplify querying
-            # the API with curl. It should allow us to do:
-            #
-            #    curl -H 'Authorization: Bearer user@education.gov.uk' http://localhost:3001/api/v2/providers
-            email = token
-          else
-            (payload, _options) = JWT.decode(token,
-                                                 Settings.authentication.secret,
-                                                 Settings.authentication.algorithm)
-            email = payload['email']
-          end
-          # match email addresses case-insensitively
+          email = get_email_from_token(token)
+
           @current_user = User.find_by("lower(email) = ?", email.downcase)
+
+          if @current_user.present? && @current_user.accept_terms_date_utc.nil?
+            head 403
```

Comment:
Will this work? If `@current_user.present?` is true then the controller action will continue as before, no?

Maybe this would be better off as a hook that triggers after the `authenticate` hook, so that it can access `@current_user` and perform it's work in isolation.

Reply:
Oh, we can actually access `@current_user` after the previous hook... that would be really good